activerecord/lib/active_record/relation/delegation.rb

```diff
@@ -89,6 +89,8 @@ def method_missing(method, *args, &block)
             self.class.delegate_to_scoped_klass(method)
             scoping { @klass.public_send(method, *args, &block) }
           elsif arel.respond_to?(method)
+            ActiveSupport::Deprecation.warn \
+              "Delegating #{method} to arel is deprecated and will be removed in Rails 6.0."
```

Comment:
The deprecation message don't appears for every call.
If `method_missing` is triggered, the `method` is defined.
So next calling don't trigger `method_missing`.

```diff
--- a/activerecord/test/cases/relation/delegation_test.rb
+++ b/activerecord/test/cases/relation/delegation_test.rb
@@ -29,6 +29,7 @@ module DeprecatedArelDelegationTests
 
     def test_deprecate_arel_delegation
       AREL_METHODS.each do |method|
+        target.public_send(method) # triggered method_missing and defined the method, then the next assertion will failed
         assert_deprecated { target.public_send(method) }
       end
     end
```